warp10/src/main/java/io/warp10/script/functions/HHCODETO.java

```diff
@@ -44,34 +48,56 @@ public HHCODETO(String name, boolean useGtsConvention) {
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
-    Object hhcode = stack.pop();
+    Object top = stack.pop();
+
+    if (top instanceof List) {
```

Comment:
This is disturbing since the conversion from a list of arguments does not use the same convention as the conversion from a single argument.

It is probably better to introduce new functions which manipulate GEOCELLS rather than piggy back on HHCODE functions.

Reply:
I made this change to mirror the behavior of `GEOHASH->`. Do you suggest removing that from both `HHCODE->` and `GEOHASH->` and making a new function of that?

Reply:
This is ok for STRING HHCodes which given their length can be interpreted as geo cells, but numerical HHCodes are different from Geo Cells which are derived from HHCodes but have a resolution prefix. Considering that a LONG appearing in a list is a Geo Cell but a LONG appearing by itself is a HHCode is what is disturbing.

Reply:
I understand what you mean.

Currently `HHCODE->` consider Long HHCodes to be of resolution 32 (so not Geo Cells) but that does not make sense to convert a list of resolution 32 HHCodes to a GeoXPShape, in fact that's not even possible without approximating.

`->HHCODELONG` converts a GeoXPShape to a list of Long GeoCell which is, as you said, mixing different concepts.

I will fix that and introduce `->GEOCELLS` and `GEOCELLS->` to convert a GeoXPShape to a list of Long or vice versa.